src/eventHandlers/JobEventHandler.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Handle events for Job.
+ */
+
+const { Op } = require('sequelize')
+const models = require('../models')
+const logger = require('../common/logger')
+const helper = require('../common/helper')
+const JobCandidateService = require('../services/JobCandidateService')
+const ResourceBookingService = require('../services/ResourceBookingService')
+
+/**
+ * Cancel all related resource bookings and all related candidates when a job is cancelled.
+ *
+ * @param {Object} payload the event payload
+ * @returns {undefined}
+ */
+async function cancelJob (payload) {
+  if (payload.value.status === payload.options.oldValue.status) {
+    logger.debug({
+      component: 'JobEventHandler',
+      context: 'cancelJob',
+      message: 'status not changed'
+    })
+    return
+  }
+  if (payload.value.status !== 'cancelled') {
+    logger.info({
+      component: 'JobEventHandler',
+      context: 'cancelJob',
+      message: `not interested job - status: ${payload.value.status}`
+    })
+    return
+  }
+  // pull data from db instead of directly extract data from the payload
+  // since the payload may not contain all fields when it is from partically update operation.
+  const job = await models.Job.findById(payload.value.id)
+  const candidates = await models.JobCandidate.findAll({
+    where: {
+      jobId: job.id,
+      status: {
+        [Op.not]: 'cancelled'
+      },
+      deletedAt: null
+    }
+  })
+  const resourceBookings = await models.ResourceBooking.findAll({
+    where: {
+      projectId: job.projectId,
+      status: {
+        [Op.not]: 'cancelled'
+      },
+      deletedAt: null
+    }
+  })
```

Comment:
@imcaizheng it looks like this logic would cancel all the ResourceBookings inside the project. While we only need to cancel ResourceBookings inside the canceled Job.

Reply:
If we can make the `jobId` an required field for the ResouceBooking model, we can then filter by `jobId` instead of `projectId`.

Reply:
@imcaizheng Got it. I would confirm if we can make `jobId` required or no. Meanwhile, let's keep it optional. So in this case we wouldn't find any ResourceBookings (if they don't have this jobId)

Reply:
@imcaizheng it looks like here we still `canceling` all the ResourceBookings inside the project, while we have to only cancel ResourceBooking associated with the particular job.